src/NuGet.Core/NuGet.DependencyResolver.Core/GraphModel/GraphOperations.cs

```diff
@@ -774,6 +817,75 @@ private static CyclesAndDowngrades CreateState(List<GraphNode<RemoteResolveResul
             };
         }
 
+        private static void DetectAndMarkAmbiguousCentralTransitiveDependencies<TItem>(Tracker<TItem> tracker, List<GraphNode<TItem>> centralTransitiveNodes)
+        {
+            // if a central transitive node has all parents disputed or ambiguous mark it and its children ambiguous
+            int ctdCount = centralTransitiveNodes.Count;
+            for (int i = 0; i < ctdCount; i++)
+            {
+                if (centralTransitiveNodes[i].Disposition == Disposition.Acceptable)
+                {
+                    bool allParentsAreDisputedOrAmbiguous = !centralTransitiveNodes[i].ParentNodes
+                        .Any(p => p.Disposition != Disposition.Rejected && !(tracker.IsDisputed(p.Item) || tracker.IsAmbiguous(p.Item)));
+
+                    if (allParentsAreDisputedOrAmbiguous)
+                    {
+                        // children of ambiguous nodes were already marked as ambiguous, skip them
+                        centralTransitiveNodes[i].ForEach(x => tracker.MarkAmbiguous(x.Item), pn => tracker.IsAmbiguous(pn.Item));
+                    }
+                }
+            };
+        }
+
+        private static void RejectCentralTransitiveBecauseOfRejectedParents<TItem>(this GraphNode<TItem> root, Tracker<TItem> tracker, List<GraphNode<TItem>> centralTransitiveNodes)
+        {
+            HashSet<GraphNode<TItem>> internalContext = new HashSet<GraphNode<TItem>>();
+
+            // reject nodes of rejected nodes for the central transitive nodes and track nodes that were not yet rejected
+            // as more nodes can be rejected do not track the nodes until all the rejects are completed
+            int ctdCount = centralTransitiveNodes.Count;
```

Comment:
It is possible that nodes that are rejected are parents of nodes and these nodes will need to be rejected and not added to the tracker. The difference in the context (tracker) is the fact that the rejected nodes will not be added to the tracker.

Reply:
> It is possible that nodes that are rejected are parents of nodes and these nodes will need to be rejected and not added to the tracker. 

This could happen sans CPVM right? Why isn't the same handling sufficient?

Reply:
> > It is possible that nodes that are rejected are parents of nodes and these nodes will need to be rejected and not added to the tracker.
> 
> This could happen sans CPVM right? Why isn't the same handling sufficient?

The difference is the fact that In case of not CPVM there will be only one parent. In case of CPVM there could be multiple parents and for this the reason keep rejecting items if all the parents were rejected and add the not rejected items to the tracker only after all the nodes with all rejected parents were handled.